### PR TITLE
[#285] Add OutgoingMessage#from_name prep

### PR DESCRIPTION
**Pull request by gbp**
## Relevant issue(s)

Connected to #285
Commits extracted from #7702 which are ready for review.

## What does this do?

Minor refactorings link to helpers and applying censor rules. These changes will make it easier to implement the #285.

### CODE REVIEW

File: spec/models/outgoing_message_spec.rb [hunk 1]
```diff
@@ -536,40 +536,6 @@ class TestError < StandardError; end
             to include("Please enter your letter requesting information")
         end
 
-        context "when a censor rule changes the default text" do
```
Comment (garethrees) [inline spec/models/outgoing_message_spec.rb:539]: Introduced in https://github.com/mysociety/alaveteli/commit/7619291e4cb7531b6d71ee6dec2dae3818c5933d; looks like it's related to this part of the conversation https://github.com/mysociety/alaveteli/pull/3270#issuecomment-225127010.

**Review by garethrees — COMMENTED** [review]
Haven't run but had a read through and LGTM in principle. Couple of questions on some of the changes.

**Comment by gbp** [comment]
I've dropped the commits with remaining issues (we will need to go into a more in depth refactoring here to clear up the remaining issues) so I can continue to make progress on https://github.com/mysociety/alaveteli/issues/285

File: app/models/outgoing_message.rb [hunk 2]
```diff
@@ -198,8 +198,9 @@ def raw_body
     read_attribute(:body)
   end
 
-  def apply_masks(text, content_type)
```
[the remark this answers is not in the document]
Comment (gbp) [reply app/models/outgoing_message.rb]: Notes for future me; We're not applying all masks to outgoing messages. Only the email mask by calling `#remove_privacy_sensitive_things!` for the HTML display (when calling `#get_body_for_html_display`) this works differently to the same method on incoming messages. We could/should? unify this into a common model concern.

File: spec/models/outgoing_message_spec.rb (hunk 1 above)
Comment (gbp) [reply spec/models/outgoing_message_spec.rb]: Notes for future me; still not convinced these specs are testing anything.

Why would the default letter need censor rules applying... for example with a censor rule doing `s/07912345678/00000000000/`, [this URL](http://0.0.0.0:3000/new/msw?default_letter=07912345678+alice@localhost.com) doesn't mask the number or email in the form textarea. Once the user previews the request the default letter isn't being used and the message body is like any other request - the number is redacted (by censor rule) and email masked as normal.

**Pull request merged**